### PR TITLE
Us 29 ii

### LINKED ISSUE #30: User Story 29 & User Story 30
(2 stories on this Issue to match Issue numbers and User Story numbers)

User Story 29, User views an Order Show Page

As a registered user
When I visit my Profile Orders page
And I click on a link for order's show page
My URL route is now something like "/profile/orders/15"
I see all information about the order, including the following information:
- the ID of the order
- the date the order was made
- the date the order was last updated
- the current status of the order
- each item I ordered, including name, description, thumbnail, quantity, price and subtotal
- the total quantity of items in the whole order
- the grand total of all items for that order

User Story 30, User cancels an order

As a registered user
When I visit an order's show page
I see a button or link to cancel the order
When I click the cancel button for an order, the following happens:
- Each row in the "order items" table is given a status of "unfulfilled"
- The order itself is given a status of "cancelled"
- Any item quantities in the order that were previously fulfilled have their quantities returned to their respective merchant's inventory for that item.
- I am returned to my profile page
- I see a flash message telling me the order is now cancelled
- And I see that this order now has an updated status of "cancelled"

### PR DESCRIPTION
**Pull request by Arique1104**
closes #30

### CODE REVIEW

File: spec/features/orders/user_order_cancel_spec.rb
Comment (lkriffell) [inline spec/features/orders/user_order_cancel_spec.rb:44]: Love this. You saved me so much time by showing me ```.relaod```

File: spec/features/orders/user_order_cancel_spec.rb
Comment (lkriffell) [inline spec/features/orders/user_order_cancel_spec.rb:46]: Just making a note that this will probably be 12 once both of our prs are merged

File: spec/features/orders/user_order_show_spec.rb
Comment (AngelaGuardia) [inline spec/features/orders/user_order_show_spec.rb:42]: Really well done with this test here Arique! Just cleaned up some whitespace.
```suggestion
    @order.item_orders.each do |item_order|
      within "#item-#{item_order.item_id}" do
        expect(page).to have_content("#{@tire.name}")
        expect(page).to have_content("#{@tire.description}")
        expect(page).to have_selector("img[src*='#{@tire.image}']")
        expect(page).to have_content("#{item_order.quantity}")
        expect(page).to have_content("#{item_order.price}")
        expect(page).to have_content("#{item_order.subtotal}")
      end
    end
```

File: app/controllers/orders_controller.rb
Comment (AngelaGuardia) [inline app/controllers/orders_controller.rb:11]: I'm guessing you needed this because we had some inconsistencies with how that id was passed into params?

I think ideally we would want to edit our routes to keep it just as :id but, this is a really cool solution you found to this! 

No need to change.

File: spec/features/orders/user_order_cancel_spec.rb
Comment (AngelaGuardia) [reply spec/features/orders/user_order_cancel_spec.rb:44]: Learned this too!

File: app/controllers/orders_controller.rb
Comment (AngelaGuardia) [inline app/controllers/orders_controller.rb:45]: Why is the else block empty? should we be flashing a message when an user tries to cancel an order that has been shipped? Maybe like this:?

```suggestion
    else
      flash[:error] = "You cannot cancel a shipped order."
    end
```

**Review by AngelaGuardia — APPROVED** [review]
Well done here @Arique. Found one empty `else` block that I think you should take a look at. Everything else looks good.

**Review by lkriffell — APPROVED** [review]
Looks good I'm glad we got those issues figured out

File: spec/features/orders/user_order_cancel_spec.rb
Comment (KieraAllen) [reply spec/features/orders/user_order_cancel_spec.rb:44]: Putting that one in my tool belt

File: spec/models/item_orders_spec.rb
Comment (KieraAllen) [inline spec/models/item_orders_spec.rb:54]: We just need to delete these comments
```suggestion
    
```

File: spec/models/item_orders_spec.rb
Comment (KieraAllen) [inline spec/models/item_orders_spec.rb:55]:
```suggestion
 
```

File: spec/features/orders/user_order_cancel_spec.rb
Comment (Arique1104) [reply spec/features/orders/user_order_cancel_spec.rb:44]: Awesome!  I honestly learned it this morning and it came in clutch!

File: app/controllers/orders_controller.rb
Comment (Arique1104) [reply app/controllers/orders_controller.rb:45]: This looks great!  I didn't want to add functionality that I wasn't requested by the user story...so I left it blank because I expected there would be an additional user story that would request some functionality that exists in the else.  But I like this....it shows great user-empathy!

File: spec/features/orders/user_order_cancel_spec.rb
Comment (Arique1104) [reply spec/features/orders/user_order_cancel_spec.rb:46]: Yay for failing tests!  Way to claim behavior you expect.  I wonder since we came up with vastly different solutions, do they impact the code in similar ways?

File: app/controllers/orders_controller.rb
Comment (Arique1104) [reply app/controllers/orders_controller.rb:11]: I think totally ideally we would want to keep things the same.  I think that because I had two different types of params coming through, this was the best way for me to be able to ensure that other params could enter while delivering on the same functionality.

But now that you mention it, maybe this was a sign that a new controller needed to be built, but I didn't build it?  Something to reflect on moving forward.

File: spec/models/item_orders_spec.rb
Comment (Arique1104) [reply spec/models/item_orders_spec.rb:54]: This is actually a failed model test.  I completely forgot about it, thank you for finding it!

**Pull request merged**
